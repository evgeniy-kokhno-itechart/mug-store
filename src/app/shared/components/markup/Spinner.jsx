import React from 'react';
import { PropTypes } from 'prop-types';

const Spinner = ({ spinnerClasses, wrapperClasses }) => (
  <div className={`text-center ${wrapperClasses}`}>
    <div className={`spinner-border ${spinnerClasses}`} role="status">
      <span className="sr-only">Loading...</span>
    </div>
  </div>
);

Spinner.propTypes = {
  spinnerClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
};

Spinner.defaultProps = {
  spinnerClasses: '',
  wrapperClasses: '',
};

export default Spinner;
