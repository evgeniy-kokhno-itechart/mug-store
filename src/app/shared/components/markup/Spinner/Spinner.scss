@import "Variables";

.spinner {
  text-align: center;
  color: $color-primary-dark;
  

  &__icon {
    animation-name: spin;
    animation-duration: 1.5s;
    animation-timing-function: linear;
    animation-iteration-count: infinite;
  }

  &--big {
    font-size: 5rem;
  }

  &--medium {
    font-size: 3rem;
  }

  &--small {
    font-size: 1rem;
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(1turn);
  }
}