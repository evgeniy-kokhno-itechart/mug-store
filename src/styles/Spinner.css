#bootstrap-overrides .spinner--big {
  width: 5rem;
  height: 5rem;
}

#bootstrap-overrides .spinner--medium {
  width: 3rem;
  height: 3rem;
}

#bootstrap-overrides .spinner--small {
  width: 1rem;
  height: 1rem;
}
