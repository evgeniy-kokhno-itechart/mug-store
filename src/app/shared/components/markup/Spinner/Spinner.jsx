import React from 'react';
import { PropTypes } from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './Spinner.scss';

const Spinner = ({ spinnerClasses, wrapperClasses }) => (
  <div className={`spinner ${wrapperClasses}`}>
    <FontAwesomeIcon icon="sync-alt" className={`spinner__icon ${spinnerClasses}`} role="status" />
    <span className="screenreaders-only">Loading...</span>
  </div>
);

Spinner.propTypes = {
  spinnerClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
};

Spinner.defaultProps = {
  spinnerClasses: '',
  wrapperClasses: '',
};

export default Spinner;
