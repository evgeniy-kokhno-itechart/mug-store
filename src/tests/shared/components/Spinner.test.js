/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Spinner from '../../../app/shared/components/markup/Spinner';

configure({ adapter: new Adapter() });

describe('<Spinner />', () => {
  let SpinnerWrapper;

  beforeEach(() => {
    SpinnerWrapper = shallow(<Spinner />);
  });

  it('renders properly with default props', () => {
    expect(SpinnerWrapper).toMatchSnapshot();
  });

  it('renders properly with spinnerClasses provided', () => {
    SpinnerWrapper.setProps({ spinnerClasses: 'testspinnerclass' });
    expect(SpinnerWrapper).toMatchSnapshot();
  });

  it('renders properly with wrapperClasses provided', () => {
    SpinnerWrapper.setProps({ wrapperClasses: 'testwrapperclass' });
    expect(SpinnerWrapper).toMatchSnapshot();
  });
});
