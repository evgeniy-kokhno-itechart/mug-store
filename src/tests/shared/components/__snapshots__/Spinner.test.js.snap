// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders properly with default props 1`] = `
<div
  className="text-center "
>
  <div
    className="spinner-border "
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`<Spinner /> renders properly with spinnerClasses provided 1`] = `
<div
  className="text-center "
>
  <div
    className="spinner-border testspinnerclass"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`<Spinner /> renders properly with wrapperClasses provided 1`] = `
<div
  className="text-center testwrapperclass"
>
  <div
    className="spinner-border "
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
